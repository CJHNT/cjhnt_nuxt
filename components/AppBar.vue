<script setup>
const { loggedIn, user, clear, fetch } = useUserSession()
const router = useRouter()

fetch()
const logout = () => {
  clear()
  router.push({ name: 'index' })
}
</script>

<template>
  <v-app-bar color="#027f42">
    <template v-slot:prepend>
      <v-btn class="d-md-none"
        ><v-img
          src="@/assets/img/cjh-logo-quadratisch-klein.png"
          alt="Logo: CJHNT Project"
          width="32"
          inline
        ></v-img
      ></v-btn>
      <v-btn
        icon="mdi-information-outline"
        href="https://cjhnt-info.saw-leipzig.de/de"
        target="_blank"
      ></v-btn>
      <v-btn icon="mdi-home" href="/"></v-btn>
    </template>
    <template v-slot:append>
      <v-tooltip location="bottom" text="Advanced Search">
        <template v-slot:activator="{ props }">
          <v-btn icon="mdi-magnify" to="/search/advanced" v-bind="props"></v-btn>
        </template>
      </v-tooltip>

      <template v-if="loggedIn">
        <v-tooltip location="bottom" :text="`Logged in as ${user.email}`">
          <template v-slot:activator="{ props }">
            <v-btn icon="mdi-account" to="/auth/dashboard" v-bind="props"></v-btn>
            <v-btn icon="mdi-logout" @click="logout" v-bind="props"></v-btn>
          </template>
        </v-tooltip>
      </template>
      <v-btn v-else icon="mdi-login" to="/auth/login"></v-btn>
      <LangSwitcher />
    </template>
  </v-app-bar>
</template>
