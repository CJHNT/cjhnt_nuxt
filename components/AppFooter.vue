<script setup>
const props = defineProps({ border: { type: String, default: '' } })
</script>

<template>
  <v-footer class="d-flex flex-column" :app="true" :border="props.border">
    <v-container class="pa-0">
      <v-row no-gutters justify="center">
        <v-col cols="auto">
          <a href="https://www.akademienunion.de/forschungsprojekte/" target="_blank">
            <v-row>
              <v-col cols="auto">
                <picture>
                  <source
                    media="(max-width: 600px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="20"
                    width="60"
                  >
                  <source
                    media="(max-width: 960px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="30"
                    width="90"
                  >
                  <source
                    media="(max-width: 1280px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="40"
                    width="120"
                  >
                  <source
                    media="(max-width: 1920px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="50"
                    width="150"
                  >
                  <source
                    media="(max-width: 2560px)"
                    srcset="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    height="60"
                    width="180"
                  >
                  <img
                    src="@/assets/img/logo_akademieunion_farbig_transparent.png"
                    alt="Logo: Akademienunion"
                    height="70"
                    width="210"
                  >
                </picture>
              </v-col>
              <v-col cols="auto">
                <picture>
                  <source
                    media="(max-width: 600px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="20"
                    width="14"
                  >
                  <source
                    media="(max-width: 960px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="30"
                    width="21"
                  >
                  <source
                    media="(max-width: 1280px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="40"
                    width="28"
                  >
                  <source
                    media="(max-width: 1920px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="50"
                    width="35"
                  >
                  <source
                    media="(max-width: 2560px)"
                    srcset="@/assets/img/sachsen-signet_gruen.png"
                    height="60"
                    width="42"
                  >
                  <img
                    src="@/assets/img/sachsen-signet_gruen.png"
                    alt="Logo: Akademienunion"
                    height="70"
                    width="47"
                  >
                </picture>
              </v-col>
              <v-tooltip activator="parent" location="top"
                >Dieses Forschungsvorhaben ist Teil im Akademienprogramm, das als derzeit größtes
                geistes- und kulturwissenschaftliches Langfrist-Forschungsprogramm der
                Bundesrepublik Deutschland von Bund und Ländern getragen wird. Diese Maßnahme wird
                mitfinanziert durch Steuermittel auf der Grundlage des vom Sächsischen Landtag
                beschlossenen Haushalts.</v-tooltip
              >
            </v-row>
          </a>
        </v-col>
      </v-row>
    </v-container>
    <div />
  </v-footer>
</template>

<style scoped>
.text-caption.footer-text {
  line-height: normal;
}
</style>

<style>
.v-tooltip__content {
  pointer-events: initial;
}
</style>
