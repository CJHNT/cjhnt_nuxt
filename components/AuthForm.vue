<script setup>
defineProps({
  loading: { type: Boolean, default: false },
  title: { type: String, required: true },
  label: { type: String, default: '' }
})
const emit = defineEmits(['submit'])
const email = ref('')
const repeatEmail = ref('')
const password = ref('')
const repeatPassword = ref('')
const showPassword = ref(false)
const showRepeatPassword = ref(false)
const wantsUpdates = ref(false)

function validateEmail() {
  const pattern =
    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  return pattern.test(email.value) || 'Invalid e-mail.'
}

const submit = () => {
  const payload = {
    email: email.value,
    repeatEmail: repeatEmail.value,
    password: password.value,
    repeatPassword: repeatPassword.value,
    wantsUpdates: wantsUpdates.value
  }
  emit('submit', payload)
}
</script>

<template>
  <v-card width="100%" class="mx-auto mt-5">
    <v-form @submit.prevent="submit">
      <v-card-title class="pb-0 text-h5">{{ $t(title) }}</v-card-title>
      <v-card-text>
        <v-text-field
          v-if="['auth.signUp', 'auth.changeEmail', 'auth.login'].includes(title)"
          v-model="email"
          :label="label ? $t(label) : 'Email'"
          :rules="[validateEmail()]"
          prepend-icon="mdi-email"
        />
        <v-text-field
          v-if="title === 'auth.changeEmail'"
          v-model="repeatEmail"
          :label="$t('auth.verifyEmail')"
          prepend-icon="mdi-email"
          :rules="[() => email === repeatEmail || $t('auth.noEmailMatch')]"
        />
        <v-text-field
          v-if="['auth.signUp', 'auth.changePassword', 'auth.login'].includes(title)"
          v-model="password"
          :type="showPassword ? 'text' : 'password'"
          :label="label ? $t(label) : $t('auth.password')"
          prepend-icon="mdi-lock"
          :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
          @click:append="showPassword = !showPassword"
        />
        <v-text-field
          v-if="['auth.signUp', 'auth.changePassword'].includes(title)"
          v-model="repeatPassword"
          :type="showRepeatPassword ? 'text' : 'password'"
          :label="$t('auth.verifyPassword')"
          prepend-icon="mdi-lock"
          :append-icon="showRepeatPassword ? 'mdi-eye' : 'mdi-eye-off'"
          :rules="[() => password === repeatPassword || $t('auth.noPasswordMatch')]"
          @click:append="showRepeatPassword = !showRepeatPassword"
        />
        <v-checkbox
          v-if="['auth.signUp'].includes(title)"
          v-model="wantsUpdates"
          :label="label ? $t(label) : $t('auth.sendMeUpdates')"
        />
      </v-card-text>
      <v-divider />
      <v-card-actions>
        <v-btn color="success" :disabled="loading" type="submit">
          <template v-if="loading">{{ $t('wait') }}</template>
          <template v-else-if="title === 'auth.changeUpdateStatus'">{{ $t(label) }}</template>
          <template v-else>{{ $t(title) }}</template>
        </v-btn>
      </v-card-actions>
      <template v-if="['auth.signUp', 'auth.login'].includes(title)">
        <v-divider />
        <v-card-text>
          <template v-if="title === 'auth.signUp'">
            {{ $t('auth.alreadyRegistered') }}
            <nuxt-link :to="{ name: 'auth-login' }">{{ $t('auth.login') }}</nuxt-link>
          </template>
          <template v-else-if="title === 'auth.login'">
            {{ $t('auth.noAccountYet') }}
            <nuxt-link :to="{ name: 'auth-register' }">{{ $t('auth.signUp') }}</nuxt-link>
          </template>
        </v-card-text>
      </template>
    </v-form>
  </v-card>
</template>
