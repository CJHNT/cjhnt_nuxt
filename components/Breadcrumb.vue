<script setup>
const props = defineProps({
  ancestors: { type: Array, required: true },
  index: { type: Number, required: true }
})
const { locale } = useI18n()
</script>

<template>
  <v-breadcrumbs :items="props.ancestors" class="py-0">
    <template v-slot:item="{ item }">
      <span v-if="item.disabled">{{ item.title[locale] }} {{ item.ref }}</span>
      <nuxt-link v-else :to="`/collection/${item.id}`">{{ item.title[locale] }}</nuxt-link>
    </template>
    <template v-slot:prepend>
      <v-icon v-if="index > 0" icon="mdi-subdirectory-arrow-right" size="small"></v-icon>
      <v-icon v-else icon="mdi-slash-forward" size="small"></v-icon>
    </template>
    <template v-slot:divider>
      <v-icon icon="mdi-slash-forward" size="small"></v-icon>
    </template>
  </v-breadcrumbs>
</template>
