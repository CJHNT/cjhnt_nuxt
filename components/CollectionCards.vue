<script setup>
const { locale } = useI18n()
const { data: theTexts } = await useAsyncData('apiAllTexts', async () => {
  const allTexts = await $fetch('http://127.0.0.1:5000/api/dts/collections', {
    query: { id: 'all_texts' }
  })
  const textPromises = allTexts.member.map(async (m) => {
    const { data: textData } = await useFetch('http://127.0.0.1:5000/api/dts/collections', {
      query: { id: m['@id'] }
    })
    const returnObject = {
      id: textData.value['@id'],
      de: textData.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'deu')[
        '@value'
      ],
      en: textData.value['dts:extensions']['dc:title'].find((e) => e['@language'] === 'eng')[
        '@value'
      ]
    }
    return returnObject
  })
  const finishedPromises = await Promise.all(textPromises)
  return finishedPromises
})
const sortedMembers = computed(() => {
  return [...theTexts.value].sort((a, b) => a.id.localeCompare(b.id))
})
import primaryImage from '@/assets/img/1_primary_texts.png'
import commentaryImage from '@/assets/img/commentary.jpg'

function getImage(urn) {
  if (urn === '1_primary_texts') {
    return primaryImage
  } else if (urn === 'urn:cts:cjhnt:commentary') {
    return commentaryImage
  }
}
</script>

<template>
  <v-row justify="center">
    <v-col v-for="member in sortedMembers" :key="member.id" cols="auto">
      <template>
        <nuxt-link class="mx-auto" :to="`/collection/${member.id}`" :title="member[locale]">
          <picture>
            <source media="(max-width: 600px)" :srcset="getImage(member.id)" height="100" />
            <source media="(max-width: 960px)" :srcset="getImage(member.id)" height="130" />
            <source media="(max-width: 1280px)" :srcset="getImage(member.id)" height="160" />
            <img :src="getImage(member.id)" :alt="member[locale]" height="200" />
          </picture>
        </nuxt-link>
      </template>
    </v-col>
  </v-row>
</template>
