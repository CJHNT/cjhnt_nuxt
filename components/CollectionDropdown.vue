<script setup>
defineProps({
  collId: {
    type: String,
    required: true,
    default: ''
  },
  collName: {
    type: String,
    required: true,
    default: ''
  }
})
</script>

<template>
  <div>
    <button>{{ collName }}</button>
    <p v-if="pending">Loading...</p>
    <CollectionDropdownContent v-else v-for="member in data.member" :key="member['@id']">{{
      member.title
    }}</CollectionDropdownContent>
  </div>
</template>
