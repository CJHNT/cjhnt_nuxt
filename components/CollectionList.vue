<script setup>
const props = defineProps({
  sortedMembers: {
    type: Array,
    required: true
  }
})
const { locale } = useI18n()
const { name } = useDisplay()
const colLists = ref([])
const numCols = ref(4)

watch(
  name,
  (newName) => {
    colLists.value = []
    let colLength = 0
    switch (newName) {
      case 'xs':
        colLists.value.push(props.sortedMembers)
        numCols.value = 1
        break
      case 'sm':
      case 'md':
        colLength = Math.ceil(props.sortedMembers.length / 3)
        for (let i = 0; i < 3; i++) {
          colLists.value.push(props.sortedMembers.slice(i * colLength, (i + 1) * colLength))
        }
        numCols.value = 3
        break
      case 'lg':
      case 'xl':
      case 'xxl':
        colLength = Math.ceil(props.sortedMembers.length / 4)
        for (let i = 0; i < 4; i++) {
          colLists.value.push(props.sortedMembers.slice(i * colLength, (i + 1) * colLength))
        }
        numCols.value = 4
        break
      default:
        colLists.value.push(props.sortedMembers)
        numCols.value = 1
    }
  },
  { immediate: true }
)
</script>

<template>
  <v-container>
    <v-row>
      <v-col v-for="(x, index) in numCols" :key="index">
        <v-list :lines="false">
          <v-list-item
            v-for="subGroup in colLists[x - 1]"
            :id="subGroup.id"
            :key="subGroup.id"
            class="pa-0"
          >
            <h2 v-if="subGroup.title" class="font-weight-bold pb-2">
              {{ $t(subGroup.title, { chapter: subGroup.chapter }) }}
            </h2>
            <template v-for="(member, memberIndex) in subGroup.subCollections" :key="memberIndex">
              <template v-if="member.versions">
                <v-list-item-title>{{ member[locale] }}</v-list-item-title>
                <v-list-item-subtitle>
                  <v-btn
                    v-for="(version, versionIndex) in member.versions"
                    :key="versionIndex"
                    :to="`/texts/${version[0]};${member.firstChild}`"
                    variant="plain"
                    density="compact"
                    >{{ version[1] }}</v-btn
                  >
                </v-list-item-subtitle>
              </template>
              <template v-else-if="member.id.includes('info')">
                <v-list-item-title>
                  <nuxt-link :to="`/texts/${member.id}`">{{
                    member[locale]
                  }}</nuxt-link></v-list-item-title
                >
              </template>
              <template v-else>
                <v-list-item-title>
                  <nuxt-link :to="`/texts/${member.id}`">{{
                    member[locale].split(' ')[0]
                  }}</nuxt-link></v-list-item-title
                >
                <v-list-item-subtitle>{{
                  member[locale].split(' ').slice(1).join(' ')
                }}</v-list-item-subtitle>
              </template>
            </template>
          </v-list-item>
        </v-list>
      </v-col>
    </v-row>
  </v-container>
</template>
