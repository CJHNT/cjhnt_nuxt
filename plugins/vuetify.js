import '@/styles/main.scss'
export default defineNuxtPlugin(() => {})
