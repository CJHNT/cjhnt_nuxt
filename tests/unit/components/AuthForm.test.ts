import { mountSuspended } from '@nuxt/test-utils/runtime'
import AuthForm from '~/components/AuthForm.vue'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'
import { createI18n } from 'vue-i18n'
import enLang from '~/lang/en.json'
import deLang from '~/lang/de.json'

const vuetify = createVuetify({
  components,
  directives
})

const i18n = createI18n({
  locale: 'en',
  messages: {
    en: enLang,
    de: deLang
  }
})

describe('AuthForm component', () => {
  test('Signup form renders correctly', async () => {
    const wrapper = await mountSuspended(AuthForm, {
      props: {
        title: 'auth.signUp'
      },
      global: {
        plugins: [vuetify, i18n]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
