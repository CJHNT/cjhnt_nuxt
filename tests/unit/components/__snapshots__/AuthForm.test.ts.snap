// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthForm component > Signup form renders correctly 1`] = `
"<div class="v-card v-theme--light v-card--density-default v-card--variant-elevated mx-auto mt-5" style="width: 100%;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background" style="opacity: NaN;"></div>
      <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <form class="v-form" novalidate="">
    <div class="v-card-title pb-0 text-h5">Sign Up</div>
    <div class="v-card-text">
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
        <div class="v-input__prepend">
          <!----><i class="mdi-email mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i>
        </div>
        <div class="v-input__control">
          <div class="v-field v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="opacity: NaN;"></div>
                <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-0">
                <!---->Email
              </label><label class="v-label v-field-label" for="input-0">
                <!---->Email
              </label>
              <!----><input size="1" type="text" id="input-0" aria-describedby="input-0-messages" class="v-field__input" value="">
              <!---->
            </div>
            <!---->
            <!---->
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <div class="v-input__details">
          <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-0-messages">
            <!---->
          </transition-group-stub>
          <!---->
        </div>
      </div>
      <!--v-if-->
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
        <div class="v-input__prepend">
          <!----><i class="mdi-lock mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i>
        </div>
        <div class="v-input__control">
          <div class="v-field v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 10px; --v-progress-linear-height: 10px;" role="progressbar" aria-hidden="false" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0">
                <!---->
                <div class="v-progress-linear__background bg-error" style="opacity: NaN;"></div>
                <div class="v-progress-linear__buffer bg-error" style="opacity: NaN; width: 0%;"></div>
                <transition-stub name="slide-x-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__determinate bg-error" style="width: 0%;"></div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-2">
                <!---->Password
              </label><label class="v-label v-field-label" for="input-2">
                <!---->Password
              </label>
              <!----><input size="1" type="password" id="input-2" aria-describedby="input-2-messages" class="v-field__input" value="">
              <!---->
            </div>
            <!---->
            <!---->
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div class="v-input__append"><i class="mdi-eye-off mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Password appended action"></i>
          <!---->
        </div>
        <div class="v-input__details">
          <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-2-messages">
            <!---->
          </transition-group-stub>
          <!---->
        </div>
      </div>
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
        <div class="v-input__prepend">
          <!----><i class="mdi-lock mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i>
        </div>
        <div class="v-input__control">
          <div class="v-field v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background" style="opacity: NaN;"></div>
                <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-4">
                <!---->Verify Password
              </label><label class="v-label v-field-label" for="input-4">
                <!---->Verify Password
              </label>
              <!----><input size="1" type="password" id="input-4" aria-describedby="input-4-messages" class="v-field__input" value="">
              <!---->
            </div>
            <!---->
            <!---->
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div class="v-input__append"><i class="mdi-eye-off mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Verify Password appended action"></i>
          <!---->
        </div>
        <div class="v-input__details">
          <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="input-4-messages">
            <!---->
          </transition-group-stub>
          <!---->
        </div>
      </div>
      <div class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-checkbox">
        <!---->
        <div class="v-input__control">
          <div class="v-selection-control v-selection-control--density-default v-checkbox-btn">
            <div class="v-selection-control__wrapper">
              <!---->
              <div class="v-selection-control__input"><i class="mdi-checkbox-blank-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input id="checkbox-6" aria-disabled="false" aria-label="Send me updates about the project" type="checkbox" aria-describedby="checkbox-6-messages" value="true"></div>
            </div><label class="v-label v-label--clickable" for="checkbox-6">
              <!---->Send me updates about the project
            </label>
          </div>
        </div>
        <!---->
        <div class="v-input__details">
          <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages" role="alert" aria-live="polite" id="checkbox-6-messages">
            <!---->
          </transition-group-stub>
          <!---->
        </div>
      </div>
    </div>
    <hr class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
    <div class="v-card-actions"><button type="submit" class="v-btn v-btn--slim v-theme--light text-success v-btn--density-default v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator="">Sign Up</span>
        <!---->
        <!---->
      </button></div>
    <hr class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
    <div class="v-card-text">Already registered? <a href="/auth/login">Login</a></div>
  </form>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
